EmberMate/AppState.swift

```diff
@@ -90,18 +99,35 @@ class AppState: ObservableObject {
                 }
             }
             .store(in: &cancellables)
+        
+        let defaults = UserDefaults.standard
+        
+        Publishers.CombineLatest(
+            defaults.publisher(for: \.notifyOnTemperatureReached).prepend(notifyOnTemperatureReached),
+            defaults.publisher(for: \.notifyOnLowBattery).prepend(notifyOnLowBattery)
+        )
+        .map { $0 || $1 }
+        .removeDuplicates()
+        .sink { notify in
+            guard notify else { return }
+            
+            self.requestNotificationAuthorization()
```

Comment:
since this is called on app load, I'm not sure its necessary to subscribe to changes

Thoughts on just calling the `UNUserNotificationCenter.current().requestAutorization(...)` process explicitly on app load, so we can cleanup some of the disjoint request locations?